server/protocols/jmap/src/main/java/org/apache/james/jmap/model/MailboxFactory.java

```diff
@@ -137,6 +140,25 @@ private Mailbox fromMessageManager(MessageManager messageManager, Optional<List<
             .build();
     }
 
+    private Quotas getQuotas(Username username) throws MailboxException {
+        QuotaRoot quotaRoot = QuotaRoot.forUser(username.getValue());
```

Comment:
Strictly speaking, I can want to distinguish QuotaRoot and user. **QuotaRoot** is a **mailbox level property**.

With such an assertion, if I untie the definition on QuotaRoot from the user, I brake it.

You should rely on the **QuotaRootResolver** to resolve the quota root from a given user.

Note: In **UserQuotaService** we should give this definition of a user quota: the quota of its INBOX mailbox. Thus we can also rely on the  QuotaRootResolver, and fully drop `QuotaRoot::forUser`.

Whn I first worked on Quota, I provided an interface for Quota bindings to let user extend this and propose arbitrary quotaRoot resolution logic. I'd like to keep this extensibility possibility... An the correct alternative is not expensive....

Reply:
Yes, I temporary used it, but it's removed in a later commit.